src/main/java/network/brightspots/rcv/ResultsWriter.java

```diff
@@ -267,54 +269,37 @@ ResultsWriter setTimestampString(String timestampString) {
   // param: precinctTallyTransfers is a map from precinct to tally transfers for that precinct
   // param: numBallotsByPrecinct is the total count of ballots per precinct
   void generatePrecinctSummaryFiles(
-      Map<String, Map<Integer, Map<String, BigDecimal>>> precinctRoundTallies,
+      Map<String, Map<Integer, RoundTally>> precinctRoundTallies,
       Map<String, TallyTransfers> precinctTallyTransfers,
-      Map<String, Integer> numBallotsByPrecinct)
+      Map<String, Integer> numUndervotesByPrecinct)
       throws IOException {
     Set<String> filenames = new HashSet<>();
     for (var entry : precinctRoundTallies.entrySet()) {
       String precinct = entry.getKey();
-      Map<Integer, Map<String, BigDecimal>> roundTallies = entry.getValue();
+      Map<Integer, RoundTally> roundTallies = entry.getValue();
       String precinctFileString = getPrecinctFileString(precinct, filenames);
       String outputPath =
           getOutputFilePathFromInstance(String.format("%s_precinct_summary", precinctFileString));
-      Integer numBallotsObj = numBallotsByPrecinct.get(precinct);
-      if (numBallotsObj == null) {
-        Logger.warning("No ballots found in precinct \"%s\"", precinct, numBallotsByPrecinct);
-        numBallotsObj = 0;
-      }
-      int numBallots = numBallotsObj;
-      generateSummarySpreadsheet(roundTallies, numBallots, precinct, outputPath);
-      generateSummaryJson(roundTallies, precinctTallyTransfers.get(precinct), precinct, outputPath);
+      Integer numUndervotes = numUndervotesByPrecinct.get(precinct);
+      generateSummarySpreadsheet(roundTallies, numUndervotes, precinct, outputPath);
+      generateSummaryJson(roundTallies, precinctTallyTransfers.get(precinct),
+            precinct, outputPath);
     }
   }
 
   // create a summary spreadsheet .csv file
   // param: roundTallies is the round-by-count count of votes per candidate
+  // param: ballotStatusTallies is the round-by-count count of ballot statuses
   // param: precinct indicates which precinct we're reporting results for (null means all)
   private void generateSummarySpreadsheet(
-      Map<Integer, Map<String, BigDecimal>> roundTallies,
-      int numBallots,
+      Map<Integer, RoundTally> roundTallies,
+      Integer numUndervotes,
```

Comment:
So has the code been using the term "undervote" inaccurately? It looks like now you're considering a ballot to be an undervote only if it doesn't contain any rankings at all. And I guess the idea is that these won't be included in the first round's results, and we'll instead present them as a single independent number? In the past I think we've also used this term to refer to cases in which a ballot skipped too many rankings.

If your definition is the right one, there are probably some more places in the code (comments, at least?) that need updating.

Reply:
You're right -- this is particularly confusing in the GUI, where there's an explicit undervote label, and that's used to label skipped rankings. To fix that, we'd need to rename config variables, update the GUI, and update a lot of comments...

Let's discuss in #633